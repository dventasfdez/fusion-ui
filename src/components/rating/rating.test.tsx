import React from "react";
import Rating from "./rating";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";

it("Rating should render and match snapshot", () => {
  const component = renderer.create(<Rating amount={10} onRate={() => null} label={{ left: "0 - Not likely", right: "Very likely - 10" }} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("On rate should be called", () => {
  const onRate = jest.fn();
  const { getByText } = render(<Rating amount={10} onRate={onRate} label={{ left: "0 - Not likely", right: "Very likely - 10" }} />);
  const button = getByText("1");
  userEvent.click(button);
  expect(onRate).toBeCalled();
});

it("Render Rating with label", () => {
  const { getByText } = render(<Rating amount={10} onRate={() => null} label={{ left: "0 - Not likely", right: "Very likely - 10" }} />);
  expect(getByText("0 - Not likely")).toBeInTheDocument();
  expect(getByText("Very likely - 10")).toBeInTheDocument();
});

it("Render Rating without label", () => {
  const { queryByText } = render(<Rating amount={10} onRate={() => null} />);
  expect(queryByText("0 - Not likely")).not.toBeInTheDocument();
});
