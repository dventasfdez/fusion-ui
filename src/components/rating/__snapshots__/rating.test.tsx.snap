// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render and match snapshot 1`] = `
<div
  className="rating "
>
  <div
    className="rating-container"
  >
    <button
      className="rating-button active"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="rating-button "
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="rating-button "
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="rating-button "
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="rating-button "
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="rating-button "
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="rating-button "
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="rating-button "
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="rating-button "
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="rating-button "
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="rating-button "
      onClick={[Function]}
      type="button"
    >
      10
    </button>
  </div>
  <div
    className="rating-label"
  >
    <small>
      0 - Not likely
    </small>
    <small>
      Very likely - 10
    </small>
  </div>
</div>
`;
